scipy/stats/tests/test_distributions.py

```diff
@@ -3419,6 +3419,54 @@ def test_isf(self):
         assert np.isclose(stats.gamma.isf(1e-50, 100),
                           330.6557590436547, atol=1e-13)
 
+class TestDgamma:
+    def test_logpdf(self):
+        x = np.array([1, 0.3, 4])
+        a = 1.3
+        y = stats.dgamma.pdf(x, a)
+        assert_allclose(y, np.exp(stats.dgamma.logpdf(x, a)))
+
+    def test_pdf(self):
+        #Reference values calculated by hand using
+        #the defintion from the Scipy documentation
+        y_1 = 0.03790773263085006
+        assert_almost_equal(y_1, stats.dgamma.pdf(-1.3, 0.3))
+        y_2 = 0.10061666529975229
+        assert_almost_equal(y_2, stats.dgamma.pdf(2.7, 4.3))
+        y_3 = 0.15705884416519492
+        assert_almost_equal(y_3, stats.dgamma.pdf(0.03, 0.01))
+
+    def test_frozen(self):
+        f = stats.dgamma(1.1)
+        assert_equal(f.pdf(1.3), stats.dgamma.pdf(1.3, 1.1))
+
+    def test_cdf(self):
+        vals = stats.dgamma.ppf([0.001, 0.5, 0.999], 1.3)
+        np.allclose([0.001, 0.5, 0.999], stats.dgamma.cdf(vals, 1.3))
+
+    def test_sf(self):
+        vals = stats.dgamma.ppf([0.001, 0.5, 0.999], 1.3)
+        S = stats.dgamma.sf(vals, 1.3)
+        assert_almost_equal(S, 1 - stats.dgamma.cdf(vals, 1.3))
+
+    def test_isf(self):
+        #Reference values calculated with mpmath
+        #import numpy as np
+        #from mpmath import mp
+        #mp.dps = 50
+
+        #def dgamma_cdf(x, a):
+        #    fac = 0.5 * mp.gammainc(a, mp.fabs(x))
+        #    return np.where(x > 0, 0.5 + fac, 0.5 - fac)
+
+        #def dgamma_isf(x, a):
+        #    return 1 / (1 - dgamma_cdf(x, a))
+
+        y_1 = 2.0033322961491538102693606074253344367238154381383
+        y_2 = 132.60693799743787520601808222040250683064041329549
+
+        np.isclose(stats.dgamma.isf(4.7, 0.01), y_1, atol = 1e-13)
+        np.isclose(stats.dgamma.isf(0.01, 1.01), y_2, atol = 1e-13)
```

Comment:
```suggestion
```
ISF is checked against SF
https://github.com/scipy/scipy/blob/0406594b35bf0b382e06d2d56ed0ac1171fbe645/scipy/stats/tests/test_continuous_basic.py#L563